lib/mongo/error/bulk_write_error.rb

```diff
@@ -35,22 +35,22 @@ class BulkWriteError < Error
       # @since 2.0.0
       def initialize(result)
         @result = result
+        super("#{self.class}: #{build_message}")
       end
 
-      def to_s
-        messages = if errors = result['writeErrors']
-          frag = ': ' + errors[0..10].map do |error|
-            "#{error['errmsg']} (#{error['code']})"
-          end.join(', ')
-          if errors.length > 10
-            frag += '...'
-          else
-            frag
-          end
-        else
-          ''
-        end
-        "#{self.class}: #{messages}" + notes_tail
+      private
+
+      def build_message
+        errors = @result['writeErrors']
+        return '' unless errors
```

Comment:
I think this should now be `return nil unless errors` since this returns the full message and if there isn't a message, this should produce nil rather than the empty string.